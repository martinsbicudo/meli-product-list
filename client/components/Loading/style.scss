$child: 0;
$bars: (
  (
    left: 0,
    animation-delay: -.30s,
    background: map-get($colors, "meli-dodger-blue")
  ),
  (
    left: 10px,
    animation-delay: -.24s,
    background: map-get($colors, "meli-gallery")
  ),
  (
    left: 20px,
    animation-delay: -.18s,
    background: map-get($colors, "meli-turbo")
  ),
  (
    left: 30px,
    animation-delay: -.12s,
    background: map-get($colors, "meli-gallery")
  ),
  (
    left: 40px,
    animation-delay: -.06s,
    background: map-get($colors, "meli-dodger-blue")
  ),
);

@keyframes accordion {

  0% {
    top: 6px;
    height: 64px;
  }

  50%,
  100% {
    top: 19px;
    height: 26px;
  }
}

.Loading {
  margin: auto;
  display: flex;
  flex: 1;
  justify-content: center;
  align-items: center;
  padding: 2rem 0;

  &Bars {
    display: inline-block;
    position: relative;
    width: 45px;
    height: 64px;
  }

  &Bar {
    display: inline-block;
    position: absolute;
    left: 7px;
    width: 5px;
    animation: accordion .5s cubic-bezier(0, .5, .5, 1) infinite;
  }
}

.LoadingBar {
  @each $bar in $bars {
    $child: $child + 1;

    &:nth-child(#{$child}) {
      left: map-get($bar, "left");
      animation-delay: map-get($bar, "animation-delay");
      background: map-get($bar, "background");
    }
  }
}
