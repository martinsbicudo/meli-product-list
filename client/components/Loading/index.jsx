import React from 'react';
import { string, func, bool } from 'prop-types';
import style from './style.scss';

const Loading = ({
  isLoading,
  children,
}) => (
  isLoading
    ? (
      <div className={style.Loading}>
        <div className={style.LoadingBars}>
          <i className={style.LoadingBar} />
          <i className={style.LoadingBar} />
          <i className={style.LoadingBar} />
          <i className={style.LoadingBar} />
          <i className={style.LoadingBar} />
        </div>
      </div>
    )
    : children()
);

Loading.propTypes = {
  message: string,
  isLoading: bool.isRequired,
  children: func.isRequired,
};

Loading.defaultProps = {
  message: '',
};

export default Loading;
