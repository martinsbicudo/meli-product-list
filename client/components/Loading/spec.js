import Loading from '.';

describe('Components - <Loading />', () => {
  const spyChildren = jest.fn().mockImplementation(() => 'TEST_CHILDREN');
  const createWrapper = (isLoading = true) => render(
    <Loading isLoading={isLoading}>{spyChildren}</Loading>,
  );

  test('Should calls onClick', () => {
    createWrapper(false);

    expect(spyChildren).toHaveBeenCalledTimes(1);
    spyChildren.mockRestore();
  });

  test('Wrapper Snapshot', () => {
    const { asFragment } = createWrapper();

    expect(asFragment()).toMatchSnapshot();
  });
});
