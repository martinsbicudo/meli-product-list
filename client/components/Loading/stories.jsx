import React from 'react';
import { storiesOf } from '@storybook/react';
import Loading from '.';

storiesOf('Loading', module)
  .add('With [isLoading=true]', () => (
    <Loading isLoading>
      {() => {}}
    </Loading>
  ))
  .add('With [isLoading=false]', () => (
    <Loading isLoading={false}>
      {() => <span>{'Doesn\'t Loading'}</span>}
    </Loading>
  ));
