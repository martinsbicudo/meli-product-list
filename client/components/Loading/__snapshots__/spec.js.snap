// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - <Loading /> Wrapper Snapshot 1`] = `
<DocumentFragment>
  <div
    class="Loading"
  >
    <div
      class="LoadingBars"
    >
      <i
        class="LoadingBar"
      />
      <i
        class="LoadingBar"
      />
      <i
        class="LoadingBar"
      />
      <i
        class="LoadingBar"
      />
      <i
        class="LoadingBar"
      />
    </div>
  </div>
</DocumentFragment>
`;
